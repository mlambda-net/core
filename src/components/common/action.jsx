import * as React from 'react';

class Action extends React.Component {
  constructor(props) {
    super(props);
    const { children } = props;
    this.state = {
      routes: React.Children.map(children, (child) => {
        return {
          name: child.props.name,
          component: child.props.children,
        };
      }),
    };
  }

  component() {
    const { routes } = this.state;
    const items = routes.filter((r) => r.name === this.props.actual);
    if (items.length === 0) {
      return <span />;
    }
    return items[0].component;
  }

  render() {
    return this.component();
  }
}

export default Action;
