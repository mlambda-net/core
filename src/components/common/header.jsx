import React from 'react';
import { AppBar, IconButton, Toolbar, Typography } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = () => ({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
});

class Header extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { children, name, classes } = this.props;
    return (
      <div className={classes.root}>
        <AppBar position="fixed" className={this.props.className}>
          <Toolbar>
            <IconButton
              style={{ display: this.props.open ? 'none' : '' }}
              edge="start"
              color="inherit"
              aria-label="menu"
              onClick={this.props.onClick}
              size="large">
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" className={classes.title}>
              {name}
            </Typography>
            {children}
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

export default withThemes(styles)(Header);
