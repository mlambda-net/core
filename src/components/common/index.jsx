export { default as Label } from './label';
export { default as LeftImage } from './leftimage';
export { default as Option } from './option';
export { default as Title } from './title';
export { default as Menu } from './menu';
export { default as Navigation } from './navigation';
export { default as Header } from './header';
export { default as withValidation } from './validation';
export { default as Action } from './action';
export { default as State } from './state';
