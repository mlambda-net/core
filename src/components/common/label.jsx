import React from 'react';
import { Box, Typography } from '@mui/material';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = () => ({
  label: {
    fontSize: '0.9em',
  },
  value: {
    fontSize: '0.8em',
  },
});

class Label extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { classes } = this.props;

    return (
      <Box style={this.props.style} className={this.props.className}>
        <Typography color="textPrimary" className={classes.label}>
          {this.props.label}
        </Typography>
        <Typography color="textSecondary" className={classes.value}>
          {this.props.value}
        </Typography>
      </Box>
    );
  }
}

export default withThemes(styles)(Label);
