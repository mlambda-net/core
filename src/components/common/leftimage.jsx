import React from 'react';
import { Box, Grid, Paper } from '@mui/material';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = () => ({});

class LeftImage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Grid container spacing={0}>
            <Grid item xs={'auto'} sm={6} height="100%">
              <Box
                height="100%"
                display="flex"
                justifyContent="center"
                alignItems="center">
                {this.props.img}
              </Box>
            </Grid>
            <Grid item xs={12} sm={6}>
              <Box
                height="100%"
                display="flex"
                justifyContent="center"
                alignItems="center">
                {this.props.children}
              </Box>
            </Grid>
          </Grid>
        </Box>
      </Paper>
    );
  }
}

export default withThemes(styles)(LeftImage);
