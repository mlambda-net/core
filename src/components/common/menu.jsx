import React from 'react';
import { Button } from '@mui/material';
import { withThemes } from '@mlambda-net/web-core/utils';

const styles = (theme) => ({
  menu: {
    border: '3px solid',
    borderColor: theme.palette.text.primary,
    textAlign: 'center',
    fontSize: theme.typography.subtitle1.fontSize,
    fontFamily: theme.typography.fontFamily,
    color: theme.palette.text.primary,
    cursor: 'pointer',
  },
});

class Menu extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <Button variant="outlined" className={classes.menu}>
        {this.props.name}
      </Button>
    );
  }
}

export default withThemes(styles)(Menu);
