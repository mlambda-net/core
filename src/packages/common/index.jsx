export Label from './label';
export LeftImage from './leftimage';
export Option from './option';
export Title from './title';
export * from './validation';
export Menu from './menu';
export Navigation from './navigation';
export Header from './header';
