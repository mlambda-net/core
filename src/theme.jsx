import { createMuiTheme } from '@material-ui/core/styles';
import pink from '@material-ui/core/colors/pink';
import teal from '@material-ui/core/colors/teal';

export const theme = createMuiTheme({
  typography: {
    fontSize: 12,
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
      '"Apple Color Emoji"',
      '"Segoe UI Emoji"',
      '"Segoe UI Symbol"',
    ].join(','),
  },
  palette: {
    primary: {
      main: pink[500],
      dark: pink[700],
      light: pink[200],
      contrastText: '#FFFFFF',
    },
    secondary: {
      main: teal[500],
      dark: teal[700],
      light: teal[200],
      contrastText: '#757575',
    },
    text: {
      primary: '#212121',
      secondary: '#757575',
    },
    divider: '#BDBDBD',
  },
});
