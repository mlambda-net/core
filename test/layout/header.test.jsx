import React from 'react';
import renderer from 'react-test-renderer';
import { expect } from '@jest/globals';
import { Header } from '@mlambda-net/core/common';
import { ThemeProvider } from '@mui/material';
import { theme } from '../theme';

test('Header render', () => {
  const item = renderer.create(
    <ThemeProvider theme={theme}>
      <Header name="app">
        <div>hello</div>
      </Header>
    </ThemeProvider>
  );
  let tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
