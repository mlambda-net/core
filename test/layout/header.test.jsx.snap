// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render 1`] = `
<div
  className="Component-root-1"
>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-mhx24p-MuiAppBar-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;width:100%;box-sizing:border-box;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;position:fixed;z-index:1100;top:0;left:auto;right:0;background-color:#1976d2;color:#fff;}@media print{.css-mhx24p-MuiAppBar-root{position:absolute;}}",
      }
    }
    data-emotion="css mhx24p-MuiAppBar-root"
  />
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-11b3ww9-MuiPaper-root-MuiAppBar-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;width:100%;box-sizing:border-box;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;position:fixed;z-index:1100;top:0;left:auto;right:0;background-color:#1976d2;color:#fff;}@media print{.css-11b3ww9-MuiPaper-root-MuiAppBar-root{position:absolute;}}",
      }
    }
    data-emotion="css 11b3ww9-MuiPaper-root-MuiAppBar-root"
  />
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-hyum1k-MuiToolbar-root{position:relative;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-left:16px;padding-right:16px;min-height:56px;}@media (min-width:600px){.css-hyum1k-MuiToolbar-root{padding-left:24px;padding-right:24px;}}@media (min-width:0px){@media (orientation: landscape){.css-hyum1k-MuiToolbar-root{min-height:48px;}}}@media (min-width:600px){.css-hyum1k-MuiToolbar-root{min-height:64px;}}",
        }
      }
      data-emotion="css hyum1k-MuiToolbar-root"
    />
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1ils2jy-MuiIconButton-root{text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;margin-left:-12px;color:inherit;padding:12px;font-size:1.75rem;}.css-1ils2jy-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-1ils2jy-MuiIconButton-root:hover{background-color:transparent;}}.css-1ils2jy-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}",
          }
        }
        data-emotion="css 1ils2jy-MuiIconButton-root"
      />
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-184mzhy-MuiButtonBase-root-MuiIconButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;margin-left:-12px;color:inherit;padding:12px;font-size:1.75rem;}.css-184mzhy-MuiButtonBase-root-MuiIconButton-root::-moz-focus-inner{border-style:none;}.css-184mzhy-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-184mzhy-MuiButtonBase-root-MuiIconButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-184mzhy-MuiButtonBase-root-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-184mzhy-MuiButtonBase-root-MuiIconButton-root:hover{background-color:transparent;}}.css-184mzhy-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}",
          }
        }
        data-emotion="css 184mzhy-MuiButtonBase-root-MuiIconButton-root"
      />
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge Component-menuButton-2 css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "display": "",
          }
        }
        tabIndex={0}
        type="button"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
            }
          }
          data-emotion="css i4bv87-MuiSvgIcon-root"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-2ulfj5-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:1.25rem;line-height:1.6;letter-spacing:0.0075em;}",
          }
        }
        data-emotion="css 2ulfj5-MuiTypography-root"
      />
      <h6
        className="MuiTypography-root MuiTypography-h6 Component-title-3 css-2ulfj5-MuiTypography-root"
      >
        app
      </h6>
      <div>
        hello
      </div>
    </div>
  </header>
</div>
`;
